extensions/socket/internal.m

```diff
@@ -0,0 +1,362 @@
+#import <LuaSkin/LuaSkin.h>
+#import <CocoaAsyncSocket/GCDAsyncSocket.h>
+
+static const char *USERDATA_TAG = "hs.socket";
+
+int refTable;
+
+@interface HSAsyncSocket : GCDAsyncSocket
+@property int callback;
+@end
+
+
+static void callback(HSAsyncSocket *asyncSocket, NSData *data) {
+    LuaSkin *skin = [LuaSkin shared];
+    NSString *utf8Data = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+
+    if (!asyncSocket.callback) {
+        lua_getglobal(skin.L, "print");
```

Comment:
I will admit I got tired of putting in callbacks in my iterations, and that I also newly discovered the ability to seamlessly patch in a function from global and thought it was cool. In my opinion there are valid uses for firing stuff without needing to read or process anything, but yea, in the case of reading, restoring the usual error might be better to more appropriately notify the user they're 'doing it wrong'.